lib/worldwide/deprecated_time_zone_mapper.rb

```diff
@@ -8,6 +10,103 @@ class << self
       def to_supported(zone)
         DEPRECATED_ZONES_MAP.fetch(zone&.to_sym, zone)
       end
+
+      # Map IANA timezones to Rails timezones
+      def to_rails(iana_timezone)
+        # If the timezone is already a supported Ruby timezone, return it
+        return iana_timezone if RAILS_TIMEZONES.key?(iana_timezone&.to_sym)
+
+        # IANA timezones has aliases. If the timezone is an alias, return the corresponding IANA timezone
+        timezone_alias = TIMEZONES_ALIASES.find do |timezone, aliases|
+          return timezone.to_s if aliases.include?(iana_timezone)
+        end
+
+        # If the alias is also a supported Rails timezone, return it
+        return timezone_alias if RAILS_TIMEZONES.key?(timezone_alias&.to_sym)
+
+        timezone = timezone_alias || iana_timezone
+
+        matching_timezones = find_matching_timezones(timezone)
```

Comment:
Timezone equivalence is more complicated than just the offset from UTC.  For example, `Africa/Casablanca` and `Africa/Monrovia` are both `UTC+0`, but [Casablanca](https://www.zeitverschiebung.net/en/city/2553604) applies DST part of the year while [Monrovia](https://www.zeitverschiebung.net/en/city/2274895) does not, so we cannot treat them as equivalent timezones.